cranelift/filetests/filetests/verifier/pointer_width_32.clif

```diff
@@ -0,0 +1,61 @@
+test verifier
```

Comment:
I introduced this file, and its 64 bit counterpart to check the verifier is now able to detect these bad cases. Most are adapted from functions in the existing clif files.

I recommend opening [this](https://github.com/bytecodealliance/wasmtime/blob/f391249b5929a41d76356f520d0bd2ab5cfb59ff/cranelift/filetests/filetests/verifier/pointer_width_32.clif) file and its [counterpart](https://github.com/bytecodealliance/wasmtime/blob/f391249b5929a41d76356f520d0bd2ab5cfb59ff/cranelift/filetests/filetests/verifier/pointer_width_64.clif) in two tabs and switching between them, they're identical save for 64 and 32 swapped.

I hope this covers all the cases we'd like to check for.